container/Dockerfile

```
@@ -19,8 +19,14 @@ RUN --mount=type=cache,target=/root/.cache/pip \
     python3 -m pip install -r /build/requirements.txt
 
 COPY . /build/karapace-repo
+WORKDIR /build/karapace-repo
 RUN --mount=type=cache,target=/root/.cache/pip \
-    python3 -m pip install /build/karapace-repo
+    if [ -z "${KARAPACE_VERSION}" ]; then \
+        PRETEND_VERSION="$(python -c 'from karapace import version; print(version.__version__)')"; \
+    else \
+        PRETEND_VERSION=$KARAPACE_VERSION; \
+    fi; \
+    SETUPTOOLS_SCM_PRETEND_VERSION=$PRETEND_VERSION python3 -m pip install --no-deps --no-cache-dir .
```

```suggestion
    SETUPTOOLS_SCM_PRETEND_VERSION=$PRETEND_VERSION python3 -m pip install --no-deps .
```

issue: The command runs with `--mount=type=cache,target=/root/.cache/pip` which sets up docker volume that is persistent between builds, so caching should work "as expected", without adding any additional weight to the resulting image.